src/main.rs

```diff
@@ -43,6 +43,34 @@ fn try_main() -> Result<()> {
         Source::from_stdin()
     };
 
+    if options.line_buffered && sources == Source::from_stdin() {
```

Comment:
Currently this silently ignores the `--line-buffered` flag when running with files which seems pretty problematic